File: go/matrices.go
package main

import "fmt"

func main() {
	matriz := [3][3]int{
		{1, 2, 3},
		{4, 5, 6},
		{7, 8, 9},
	}
	fmt.Println("Matriz inicial:", matriz)

	fmt.Println("\n Imprimir la matriz en forma de tabla")
	for i := 0; i < len(matriz); i++ {
		for j := 0; j < len(matriz[i]); j++ {
			fmt.Printf("%d\t", matriz[i][j])
		}
		fmt.Println()
	}

	fmt.Println("\n Recorrer por columnas")
	filas := len(matriz)
	columnas := len(matriz[0])
	for col := 0; col < columnas; col++ {
		for fila := 0; fila < filas; fila++ {
			fmt.Printf("%d\t", matriz[fila][col])
		}
		fmt.Println()
	}

	fmt.Println("\n Sumar todos los elementos")
	suma := 0
	for i := 0; i < len(matriz); i++ {
		for j := 0; j < len(matriz[i]); j++ {
			suma += matriz[i][j]
		}
	}
	fmt.Println("Suma total:", suma)

	fmt.Println("\n Intercambiar la primera fila con la última")
	ultima := len(matriz) - 1
	matriz[0], matriz[ultima] = matriz[ultima], matriz[0]
	fmt.Println("Matriz con filas intercambiadas:", matriz)
}

File: javascript/matrices.js
let matriz = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9],
];
console.log("Matriz inicial:");
console.log(matriz);

console.log("\n Imprimir la matriz en forma de tabla");
for (let i = 0; i < matriz.length; i++) {
    let fila = "";
    for (let j = 0; j < matriz[i].length; j++) {
        fila += matriz[i][j] + "\t";
    }
    console.log(fila);
}

console.log("\n Recorrer por columnas");
const filas = matriz.length;
const columnas = matriz[0].length;
for (let col = 0; col < columnas; col++) {
    let columna = "";
    for (let fila = 0; fila < filas; fila++) {
        columna += matriz[fila][col] + "\t";
    }
    console.log(columna);
}


console.log("\n Sumar todos los elementos");
let suma = 0;
for (let i = 0; i < matriz.length; i++) {
    for (let j = 0; j < matriz[i].length; j++) {
        suma += matriz[i][j];
    }
}
console.log("Suma total:", suma);

console.log("\n Intercambiar la primera fila con la última");
const ultima = matriz.length - 1;
[matriz[0], matriz[ultima]] = [matriz[ultima], matriz[0]];
console.log("Matriz con filas intercambiadas:");
console.log(matriz);